contrib/format-fixedwidth/src/main/java/org/apache/drill/exec/store/fixedwidth/FixedwidthBatchReader.java

```diff
@@ -0,0 +1,193 @@
+import org.apache.drill.exec.physical.resultSet.ResultSetLoader;
+import org.apache.drill.exec.physical.resultSet.RowSetLoader;
+import org.apache.drill.exec.record.metadata.SchemaBuilder;
+import org.apache.drill.exec.record.metadata.TupleMetadata;
+import org.apache.drill.shaded.guava.com.google.common.base.Charsets;
+import org.apache.hadoop.mapred.FileSplit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+
+public class FixedwidthBatchReader implements ManagedReader<FileSchemaNegotiator>{
+
+  private static final Logger logger = LoggerFactory.getLogger(FixedwidthBatchReader.class);
+
+  private FileSplit split;
+
+  private final int maxRecords;
+
+  private final FixedwidthFormatConfig config;
+
+  private CustomErrorContext errorContext;
+
+  private InputStream fsStream;
+
+  private ResultSetLoader loader;
+
+  private BufferedReader reader;
+
+  public FixedwidthBatchReader(FixedwidthFormatConfig config, int maxRecords) {
+    this.config = config;
+    this.maxRecords = maxRecords;
+  }
+
+  @Override
+  public boolean open(FileSchemaNegotiator negotiator) {
+    split = negotiator.split();
+    errorContext = negotiator.parentErrorContext();
+    try {
+      fsStream = negotiator.fileSystem().openPossiblyCompressedStream(split.getPath());
+      negotiator.tableSchema(buildSchema(),true);
+      loader = negotiator.build();
+    } catch (Exception e) {
+      throw UserException
+              .dataReadError(e)
+              .message("Failed to open input file: {}", split.getPath().toString())
+              .addContext(errorContext)
+              .addContext(e.getMessage())
+              .build(logger);
+    }
+
+    reader = new BufferedReader(new InputStreamReader(fsStream, Charsets.UTF_8));
+
+    return true;
+
+  }
+
+  @Override
+  public boolean next() { // Use loader to read data from file to turn into Drill rows
+
+    String line;
+
+    try {
+      line = reader.readLine();
+      RowSetLoader writer = loader.writer();
+
+      while (!writer.isFull() && line != null) {
+
+        writer.start();
+        parseLine(line, writer);
+        writer.save();
+
+        line = reader.readLine();
+      }
+    } catch (Exception e) {
+      throw UserException
+              .dataReadError(e)
+              .message("Failed to read input file: {}", split.getPath().toString())
+              .addContext(errorContext)
+              .addContext(e.getMessage())
+              .build(logger);
+    }
+    return (line != null);
+  }
+
+  @Override
+  public void close() {
+    try {
+      fsStream.close();
```

Comment:
Consider using `Autoclosables` here.  That will do all the error handling for you. 

```java
 @Override
  public void close() {
    if (fsStream != null) {
      AutoCloseables.closeSilently(fsStream);
      fsStream = null;
    }
  }
```

Reply:
Done